### PR TITLE
fix test failure

### PR DESCRIPTION
**Pull request by Pearl1594**
## Description
test_ssvm fails on VMWare fixes: https://github.com/apache/cloudstack/issues/4261
Running the ssvm-check.sh script on the Secondary storage VM produces the following error message:
```
command substitution: ignored null byte in input
```
as null characters were being passed, which caused the test to fail. 

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@Pearl1594 jfyi @spaceman1984 is looking at vm migration failure. Also it's possible the issue is on master and not 4.14, pl check that as well

**Comment by Pearl1594** [comment]
@blueorangutan test

**Comment by Pearl1594** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan test centos7 vmware-67u3

File: systemvm/agent/scripts/ssvm-check.sh [hunk 1]
```diff
@@ -118,7 +118,7 @@ fi
 # check for connectivity to the management server
 echo ================================================
 echo Management server is $MGMTSERVER.  Checking connectivity.
-socatout=$(echo | socat - TCP:$MGMTSERVER:8250,connect-timeout=3 2>&1)
+socatout=$(echo | socat - TCP:$MGMTSERVER:8250,connect-timeout=3 | tr -d '\0' 2>&1)
```
Comment (rhtyd) [inline systemvm/agent/scripts/ssvm-check.sh:121]: @Pearl1594 what was the issue during the reboot test, does this fix that? If so, do we need to port it back to 4.14/4.13?

Comment (Pearl1594) [reply systemvm/agent/scripts/ssvm-check.sh:121]: while it reports a failure in ssvm reboot test, the logs report a failure in ssvm check:
```
2020-08-11 23:39:12,513 - CRITICAL - FAILED: test_07_reboot_ssvm: ['Traceback (most recent call last):\n', '  File "/usr/lib64/python2.7/unittest/case.py", line 369, in run\n    testMethod()\n', '  File "/marvin/tests/smoke/test_ssvm.py", line 897, in test_07_reboot_ssvm\n    self.test_03_ssvm_internals()\n', '  File "/marvin/tests/smoke/test_ssvm.py", line 460, in test_03_ssvm_internals\n    "Check for Errors in tests"\n', '  File "/usr/lib64/python2.7/unittest/case.py", line 553, in assertEqual\n    assertion_func(first, second, msg=msg)\n', '  File "/usr/lib64/python2.7/unittest/case.py", line 546, in _baseAssertEqual\n    raise self.failureException(msg)\n', 'AssertionError: Check for Errors in tests\n']
```

**Comment by Pearl1594** [comment]
Seems like, while sometimes the issue is wrt to the running of socat command, during reboot, it fails because of the following error:
ERROR: NFS is not currently mounted.

**Comment by rhtyd** [comment]
@Pearl1594 I see the issue as well, I've done repeated tests on master now:
```
test_07_reboot_ssvm (tests.smoke.test_ssvm.TestSSVMs): DEBUG: SSVM script output: [u'ERROR: NFS is not currently mounted', u'Tests Complete. Look for ERROR or WARNING above.']
```

Can you investigate why NFS is not mounted or should it be? Should we do something to make it mount for example add a template or download a template, or simply ignore this case? You may use the lab env `pr4131-t2412-vmware-67u3`

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan test centos7 vmware-67u3

**Pull request merged**